src/main/java/springbootquickstarter/services/CourseService.java

```diff
@@ -14,43 +17,56 @@
 public class CourseService {
 	@Autowired
 	private CourseRepository courseRepository;
-	
-
-public List <Course> getAllCourses(Long topicId){
-	List <Course> topics= new ArrayList<>();
-	courseRepository.findByTopicId(topicId)
-	.forEach(topics::add);
-	return topics;
-}
-public ResponseEntity<Course> getCourse(Long id) throws CourseDoesNotExist {
-	 return ResponseEntity.ok(findCourse(id));
-}
-public ResponseEntity<Course> addCourse(Course course) {
-	return ResponseEntity.ok(courseRepository.save(course));	
-}
-public ResponseEntity<Course> updateCourse(Course course) throws CourseDoesNotExist {
-	return ResponseEntity.ok(courseRepository.save(findCourse(course.getId())));
-}
-public ResponseEntity<Void> deleteCourse(Long id) throws CourseDoesNotExist {
-	courseRepository.delete(findCourse(id).getId());
-	return ResponseEntity.ok().build();	
-}
-public List <Student> getCourseStudents(Long id) throws CourseDoesNotExist {
- return findCourse(id).getStudents();
-}
-public ResponseEntity<Course> addStudentToCourse(Long courseid, Student students) throws CourseDoesNotExist {
-	Course c = findCourse(courseid);
-	c.setStudents(students);
-	return ResponseEntity.ok(courseRepository.save(c));
-}
-
-public Course findCourse(Long id) throws CourseDoesNotExist {
-	Course course = courseRepository.findOne(id);
-	if (course != null) {
-		return course;
-	}
-	throw new CourseDoesNotExist("Course of id = " + id + " does not exist.");
-}
 
+	public List<Course> getAllCourses(Long topicId) {
+		List<Course> topics = new ArrayList<>();
+		courseRepository.findByTopicId(topicId).forEach(topics::add);
+		return topics;
+	}
+
+	public ResponseEntity<Course> getCourse(Long id) throws CourseDoesNotExist {
+		return ResponseEntity.ok(findCourse(id));
+	}
+
+	public ResponseEntity<Course> addCourse(Course course) {
+		return ResponseEntity.ok(courseRepository.save(course));
+	}
+
+	public ResponseEntity<Course> updateCourse(Course course) throws CourseDoesNotExist {
+		return ResponseEntity.ok(courseRepository.save(findCourse(course.getId())));
+	}
+
+	public ResponseEntity<Void> deleteCourse(Long id) throws CourseDoesNotExist {
+		courseRepository.delete(findCourse(id).getId());
+		return ResponseEntity.ok().build();
+	}
+
+	public List<Student> getCourseStudents(Long id) throws CourseDoesNotExist {
+		return findCourse(id).getStudents();
+	}
+
+	public ResponseEntity<Course> addStudentToCourse(Long courseid, Student students) throws CourseDoesNotExist {
+		Course c = findCourse(courseid);
+		c.setStudents(students);
+		return ResponseEntity.ok(courseRepository.save(c));
+	}
+
+	public Course findCourse(Long id) throws CourseDoesNotExist {
+		Course course = courseRepository.findOne(id);
+		if (course != null) {
+			return course;
+		}
+		throw new CourseDoesNotExist("Course of id = " + id + " does not exist.");
+	}
+
+	public List<Student> getCoursesStudents(Map<String, String> my_map) throws CourseDoesNotExist {
+		List<Student> students = new ArrayList<>();
```

Comment:
this is should be Set, since there will be duplicate students. 
also is there another way to do it in one mysql query?

Reply:
Set is modified. 
About the query, do you mean something other than the Map? Or is it about the usage of the method? 
I found that I have already implemented the find students for a single course, so why not to use it?